pallets/cf-reward/src/lib.rs

```diff
@@ -0,0 +1,208 @@
+    fn reward() -> Weight { 0 }
+}
+
+
+decl_event!{
+    pub enum Event<T> where AccountId = <T as frame_system::Config>::AccountId {
+        /// Event emitted when a reward claim was processed successfully.
+        RewardClaimed(AccountId),
+        /// Event triggered when the reward module is ready to reward contributors
+        RewardModuleInitalized,
+    }
+}
+
+decl_storage! {
+    trait Store for Module<T: Config> as Reward {
+        NextInit get(fn next_init) config(): T::BlockNumber;
+        ModuleAccount get(fn mod_account) : T::AccountId = MODULE_ID.into_account();
+        ConversionRate get(fn conversion_rate) config(): BalanceOf<T>;
+    }
+}
+
+decl_error! {
+	pub enum Error for Module<T: Config> {
+        /// Not enough funds in the pot for paying a reward
+        NotEnoughFunds,
+        /// Initalization before the last lease period is over
+        OngoingLease,
+    }
+}
+
+
+decl_module! {
+    pub struct Module<T: Config> for enum Call where origin: <T as frame_system::Config>::Origin {
+        // Activate errors
+        type Error = Error<T>;
+
+        // Activate events
+        fn deposit_event() = default;
+
+        /// Initialize the module.
+        /// This dispatchable function must be called from a root account, so that other accounts
+        /// can not block the module via a random initialization. Furthermore it is important, that
+        /// the module has enough funds, as a later funding is not possible.
+        // TODO: Add correct weight function
+        #[weight = 10_000]
+        fn initialize(origin, source: T::AccountId, amount: BalanceOf<T>) -> DispatchResult {
+            ensure_root(origin)?;
+
+            let now = <frame_system::Module<T>>::block_number();
+            ensure!(now > <NextInit<T>>::get(), Error::<T>::OngoingLease);
+
+            let target = &MODULE_ID.into_account();
+            // TODO: Should we choose: KeepAlive or AllowDeath here?
+            T::Currency::transfer(
+                &source,
+                &target,
+                amount,
+                AllowDeath,
+            )?;
+
+            <NextInit<T>>::put(now + T::LeasePeriod::get());
+
+            Self::deposit_event(RawEvent::RewardModuleInitalized);
+
+            Ok(())
+        }
+    }
+}
+
+impl<T: Config> Module<T> {
+    fn convert_to_native(contribution: T::RelayChainBalance) -> <T::Currency as Currency<T::AccountId>>::Balance {
+        contribution.into() * <ConversionRate<T>>::get()
+    }
+}
+
+impl<T: Config> Reward for Module<T> {
+    type ParachainAccountId = T::AccountId;
+    type ContributionAmount = T::RelayChainBalance;
+
+    fn reward(who: Self::ParachainAccountId, contribution: Self::ContributionAmount) -> DispatchResult {
+        let schedule = pallet_vesting::VestingInfo {
+            locked: Self::convert_to_native(contribution),
+            per_block: <<T as pallet_vesting::Config>::BlockNumberToBalance>::convert(T::VestingPeriod::get()),
```

Comment:
PerBlock here means the amount of tokens unlocked per block. Therefore that depends on each contribution amount and for how long are we vesting (LeasePeriod). 
So we need to change that to `per_block = contribution / n_total_blocks` and `n_total_blocks = LeasePeriod` I believe.